files/image_config/platform/rc.local

```diff
@@ -79,43 +79,25 @@ if [ -f /host/image-$sonic_version/platform/firsttime ]; then
         firsttime_exit
     fi
 
-    # setup initial switch mac
-    SONIC_ASIC_TYPE=$(sonic-cfggen -y /etc/sonic/sonic_version.yml -v asic_type)
-    SYSTEM_MAC_ADDRESS=$(ip link show eth0 | grep ether | awk '{print $2}')
-
-    # Align last byte of MAC if necessary
-    if [ "$SONIC_ASIC_TYPE" = "mellanox" ] || [ "$SONIC_ASIC_TYPE" = "centec" ]; then
-        last_byte=$(python -c "print '$SYSTEM_MAC_ADDRESS'[-2:]")
-        aligned_last_byte=$(python -c "print format(int(int('$last_byte', 16) & 0b11000000), '02x')")  # put mask and take away the 0x prefix
-        SYSTEM_MAC_ADDRESS=$(python -c "print '$SYSTEM_MAC_ADDRESS'[:-2] + '$aligned_last_byte'")      # put aligned byte into the end of MAC
-    fi
-
-    if [ -f /etc/sonic/init_cfg.json ]; then
-        sonic-cfggen -j /etc/sonic/init_cfg.json -a '{"DEVICE_METADATA":{"localhost": {"mac": "'$SYSTEM_MAC_ADDRESS'"}}}' --print-data > /tmp/init_cfg.json
-        mv /tmp/init_cfg.json /etc/sonic/init_cfg.json
-    else
-        sonic-cfggen -a '{"DEVICE_METADATA":{"localhost": {"mac": "'$SYSTEM_MAC_ADDRESS'"}}}' --print-data > /etc/sonic/init_cfg.json
-    fi
-
     # Try to take old configuration saved during installation
     if test_config; then
         rm -f /host/old_config/sonic_version.yml
         mv -f /host/old_config/* /etc/sonic/
         if [ ! -f /etc/sonic/config_db.json ]; then
-            sonic-cfggen -m -j /etc/sonic/init_cfg.json --print-data > /etc/sonic/config_db.json
+            sonic-cfggen -H -m -j /etc/sonic/init_cfg.json --print-data > /etc/sonic/config_db.json
```

Comment:
do we still need init_cfg.json? here, what if init_cfg.json does not exist? will it have issue here?

Reply:
Thanks for the catch. Fixed in the 2nd iteration.
Please note that this PR does not intent to make any logic change - I will submit a PR later to remove graph-selecting and config-loading logic from rc.local and unify them in update_graph.